File: tools/batch-srv/src/Flowbox/Batch/Server/Handler/Process.hs
---------------------------------------------------------------------------
-- Copyright (C) Flowbox, Inc - All Rights Reserved
-- Unauthorized copying of this file, via any medium is strictly prohibited
-- Proprietary and confidential
-- Flowbox Team <contact@flowbox.io>, 2013
---------------------------------------------------------------------------
module Flowbox.Batch.Server.Handler.Process where

import Data.IORef (IORef)

import qualified Data.IORef                                     as IORef
import           Flowbox.Batch.Batch                            (Batch)
import qualified Flowbox.Batch.Handler.Process                  as BatchP
import           Flowbox.Prelude
import           Flowbox.System.Log.Logger
import           Flowbox.Tools.Serialize.Proto.Conversion.Basic
import qualified Generated.Proto.Batch.Process.Processes.Args   as Processes
import qualified Generated.Proto.Batch.Process.Processes.Result as Processes
import qualified Generated.Proto.Batch.Process.Terminate.Args   as Terminate
import qualified Generated.Proto.Batch.Process.Terminate.Result as Terminate



loggerIO :: LoggerIO
loggerIO = getLoggerIO "Flowbox.Batch.Server.Handlers.Process"

-------- public api -------------------------------------------------

processes :: IORef Batch -> Processes.Args -> IO Processes.Result
processes batchHandler (Processes.Args tprojectID) = do
    loggerIO info "called processes"
    let projectID = decodeP tprojectID
    batch <- IORef.readIORef batchHandler
    processIDs <- BatchP.processes projectID batch
    return $ Processes.Result $ encodeListP processIDs


terminate :: IORef Batch -> Terminate.Args -> IO Terminate.Result
terminate batchHandler (Terminate.Args tprocessID tprojectID) = do
    loggerIO info "called terminate"
    let processID = decodeP tprocessID
        projectID = decodeP tprojectID
    batch <- IORef.readIORef batchHandler
    newBatch <- BatchP.terminate processID projectID batch
    IORef.writeIORef batchHandler newBatch
    return Terminate.Result

File: scripts/utils/process.py
###########################################################################
## Copyright (C) Flowbox, Inc / All Rights Reserved
## Unauthorized copying of this file, via any medium is strictly prohibited
## Proprietary and confidential
## Flowbox Team <contact@flowbox.io>, 2014
###########################################################################

import sys

from colors import print_error
from errors import handle_out
from subprocess import call
from utils.colors      import print_error, formatQuestion

def autocall(cmd_argv):
    handle_out(call(cmd_argv))

def ask(question, silent=False):
    if silent:
        print(formatQuestion("Question override: " + question))
        return True
    else:
        answer = None
        while True:
            ansRaw = raw_input(formatQuestion(question) + " [Y/n] ")
            ans = ansRaw.lower()
            if ans not in ["", "y", "yes", "n", "no"]:
                print_error("'%s' is not a valid answer, please choose beetween YES or NO." % ansRaw)
            else: break
        if ans in ["", "y", "yes"]: return True
        else:                       return False